lib/spack/spack/cmd/list.py

```diff
@@ -113,9 +114,16 @@ def match(p, f):
                 return f.match(p)
 
         pkgs = [p for p in pkgs if any(match(p, f) for f in res)]
+        
+    # If tags have been specified on the command line, filter by tags
+    if args.tags:
+        packages_with_tags = spack.repo.path.packages_with_tags(*args.tags)
+        results = [x for x in results if x.name in packages_with_tags]    
```

Comment:
```suggestion
        pkgs = [p for p in pkgs if p.name in packages_with_tags]    
```
The results variable isn't defined or used anywhere in this function